main.py

```diff
@@ -2293,89 +2296,168 @@ async def ping(message: discord.Interaction):
+                button2.callback = evade
+    
+                myview = View(timeout=3600)
+    
+                myview.add_item(button)
+                myview.add_item(button2)
+    
+                await message.response.send_message(embed=embed, view=myview)
+            else:
+                await message.response.send_message(embed=embed)
+    
+            # handle aches
+            await achemb(message, "donator", "send")
+            await achemb(message, "anti_donator", "send", person)
+            if person_id == bot.user.id and cat_type == "Ultimate" and int(amount) >= 5:
+                await achemb(message, "rich", "send")
+            if person_id == bot.user.id:
+                await achemb(message, "sacrifice", "send")
+            if cat_type == "Nice" and int(amount) == 69:
+                await achemb(message, "nice", "send")
+        else:
+            await message.response.send_message("no", ephemeral=True)
+    elif cat_type.lower() == "rain":
+        if person_id == bot.user.id:
+            await message.response.send_message("you can't sacrifice rains", ephemeral=True)
+            return
 
-            myview = View(timeout=3600)
+        actual_user, _ = User.get_or_create(user_id=message.user.id)
+        actual_receiver, _ = User.get_or_create(user_id=person_id)
+        if actual_user.rain_minutes >= amount:
+            actual_user.rain_minutes -= amount
+            actual_receiver.rain_minutes += amount
+            actual_user.save()
+            actual_receiver.save()
+            embed = discord.Embed(title="Success!", description=f"Successfully transfered {amount:,} minutes of rain from <@{message.user.id}> to <@{person_id}>!", color=0x6E593C)
+
+            # handle tax
+            if amount >= 5:
+                tax_amount = round(amount * 0.2)
+
+                async def confirm_pay(interaction):
+                    if interaction.user.id == message.user.id:
+                        await interaction.response.defer()
+                        confirm = Button(label="Yes, pay the tax")
+                        confirm.callback = pay
+                        confirm_view = View(timeout=3600)
+                        confirm_view.add_item(confirm)
+                        await interaction.followup.send(f"Are you really sure you want to pay the tax? {tax_amount:,} minutes of rain will be lost forever...", view=confirm_view)
+                    else:
+                        await do_funny(interaction)
 
-            myview.add_item(button)
-            myview.add_item(button2)
+                async def pay(interaction):
+                    if interaction.user.id == message.user.id:
+                        await interaction.response.defer()
+                        try:
+                            await interaction.edit_original_response(view=None)
+                        except Exception:
+                            pass
+                        actual_user.rain_minutes -= tax_amount
```

Comment:
get the user again before doing that to prevent weird shit

Reply:
okay now i realize its not done for cat taxes either so maybe its fine?
i think its safer to also add it there then

Reply:
ok i'll look at it later, don't think there is an issue caused by that though